.github/workflows/continuous_integration.yml

```diff
@@ -0,0 +1,170 @@
+
+          - name: Molecular Subtyping - LGG
+            entrypoint: molecular-subtyping-LGAT/run_subtyping.sh
+            openpbta_subset: 0
+
+          - name: Molecular Subtyping - PATHOLOGY/COMPILE
+            entrypoint: molecular-subtyping-pathology/run-subtyping-aggregation.sh
+            openpbta_testing: 1
+
+          - name: Molecular Subtyping - INTEGRATE (+ add cancer groups)
+            entrypoint: molecular-subtyping-integrate/run-subtyping-integrate.sh
+
+          # Analysis modules
+          - name: Independent Specimens
+            entrypoint: independent-samples/run-independent-samples.sh
+
+          - name: Independent Specimens pre-release
+            entrypoint: independent-samples/run-independent-samples.sh
+            run_for_subtyping: 1
+
+          - name: TP53/NF1 scores
+            entrypoint: tp53_nf1_score/run_classifier.sh
+            openpedcan_polya_strand: 0
+
+          - name: Fusion filtering
+            entrypoint: fusion_filtering/run_fusion_merged.sh
+
+          - name: Fusion summary
+            entrypoint: fusion-summary/run-new-analysis.sh
+            openpbta_subset: 0
+
+          - name: Consensus CN
+            entrypoint: focal-cn-file-preparation/run-prepare-cn-OpenTarget.sh
+            openpbta_testing: 1
+
+          - name: TMB calculation
+            entrypoint: tmb-calculation/run_tmb_calculation.sh
+
+          - name: Immune Deconvolution
+            entrypoint: immune-deconv/run-immune-deconv.sh
+
+          - name: EFO/MONDO annotation
+            entrypoint: efo-mondo-mapping/run_qc.sh
+
+          - name: ENSEMBL Gene matching
+            entrypoint: gene_match/run-gene-mapping.sh
+
+          - name: Update table annotation data
+            entrypoint: long-format-table-utils/run-update-long-format-table-utils.sh
+
+          - name: Fusion frequency tables
+            entrypoint: fusion-frequencies/run-frequencies.sh
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Cache Data
+        uses: actions/cache@v3
+        with:
+          path: ./data
+          key: ${{ runner.os }}-data-${{ hashFiles('download-data.sh') }}
+          restore-keys: |
+            ${{ runner.os }}-data-
+
+      - name: Download Data
+        uses: docker://pgc-images.sbgenomics.com/d3b-bixu/open-pedcan:analysisjob
+        with:
+          entrypoint: ./download-data.sh
+        env:
+          OPENPEDCAN_URL: https://s3.amazonaws.com/d3b-openaccess-us-east-1-prd-pbta/open-targets
+          OPENPEDCAN_RELEASE: testing
+
+      - name: Run Analysis
+        uses: docker://pgc-images.sbgenomics.com/d3b-bixu/open-pedcan:analysisjob
+        with:
+          entrypoint: analyses/${{ matrix.entrypoint }}
+        env:
+          OPENPBTA_SUBSET: ${{ matrix.openpbta_subset }}
+          OPENPBTA_TESTING: ${{ matrix.openpbta_testing }}
+          RUN_FOR_SUBTYPING: ${{ matrix.run_for_subtyping }}
+          OPENPEDCAN_POLYA_STRAND: ${{ matrix.openpedcan_polya_strand }}
```

Comment:
If one of the matrix items doesn't define a variable, like `openpbta_subset`, then GitHub Actions will pass an empty string to the analysis.

Testing shows that the analysis shell scripts will use their default values if provided an empty string:

```console
$ OPENPBTA_SUBSET=""             
$ SUBSET=${OPENPBTA_SUBSET:-1}
$ echo $SUBSET
1
```

Reply:
good to know, and if we add additional variables later, we will have to add here as well, correct @rbreslow ?

Reply:
@jharenza to circle back on this, yes we can add them here in the future, I think they can also be run in-line.